### PR TITLE
Fix `tests/sanity/routing_table_sync.py`test

### PR DESCRIPTION
**Pull request by pmnoxx**
`tests/sanity/routing_table_sync.py` test wasn't fully fixed.

We were not pruning while `prune == Prune::PruneNow`, because no routing table calculation was done.

This PR fixed the test properly.

The bug was in not having ```prune == Prune::PruneNow ``` check. In this PR we will improve code quality as well, and split `recalculate_routing_table_and_maybe_prune_edges` into to separate methods.

### CODE REVIEW

**Comment by pmnoxx** [comment]
@mm-near @bowenwang1996 Can you take a look?

File: chain/network/src/routing/routing_table_actor.rs [hunk 1]
```diff
@@ -511,23 +516,21 @@ impl Handler<RoutingTableMessages> for RoutingTableActor {
...
-                    self.recalculate_routing_table_and_maybe_prune_edges(
-                        prune,
-                        prune_edges_not_reachable_for,
-                    )
-                } else {
-                    Vec::new()
-                };
-                self.needs_routing_table_recalculation = false;
+                let mut edges_removed =
+                    if self.needs_routing_table_recalculation || prune == Prune::PruneNow {
```
Comment (mm-near) [inline chain/network/src/routing/routing_table_actor.rs:520]: so this is the place where the bug was, right?

and the refactoring/splitting of the methods was just to improve the readability?
(if yes - please mention it in PR description - as I was looking for the bug in those methods instead)

**Review by mm-near — APPROVED** [review]

File: chain/network/src/routing/routing_table_actor.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/network/src/routing/routing_table_actor.rs:520]: Thanks. Yes, that was the bug. I updated the PR.

**Comment by bowenwang1996** [comment]
Can we actually attach a link to a nayduck run that runs the test 30 times to verify the test is indeed fixed?

**Comment by pmnoxx** [comment]
> Can we actually attach a link to a nayduck run that runs the test 30 times to verify the test is indeed fixed?

How can I schedule a single test to run on nayduck?

**Comment by pmnoxx** [comment]
@bowenwang1996 Ok, I'll investigate on how to run single test on nayduck.

**Comment by pmnoxx** [comment]
> Can we actually attach a link to a nayduck run that runs the test 30 times to verify the test is indeed fixed?

Thanks, that a good suggestion.

**Pull request merged**